src/as_list.cpp

```diff
@@ -0,0 +1,34 @@
+#include "r_trie.h"
+
+template <typename T>
+static inline Rcpp::List trie_to_list(SEXP radix){
+  r_trie <T>* rt_ptr = (r_trie <T> *) R_ExternalPtrAddr(radix);
+  ptr_check(rt_ptr);
+  
+  return Rcpp::List::create(
+    Rcpp::Named("keys") = rt_ptr->get_keys(),
```

Comment:
...well that's embarrassing. Yes, your solution is much better, and much simpler. I blame all this C++ for rotting my brain.